operator-pipeline-images/operatorcert/entrypoints/request_signature.py

```diff
@@ -192,44 +203,92 @@ def request_signature(args: Any) -> None:
     """
     Format and send out a UMB message to request signing, and retry as needed.
     """
-    global umb
+    manifests = args.manifest_digest.split(",")
+    references = args.reference.split(",")
+
+    if len(manifests) != len(references):
+        LOGGER.error(
+            "Manifest digest list does not match the length of reference list."
+        )
+        sys.exit(1)
+
     umb = start_umb_client(
         hosts=[args.umb_url], client_name=args.umb_client_name, handler=UmbHandler()
     )
     global result_file
     result_file = args.output
 
-    request_msg = gen_request_msg(args, umb.id)
+    request_msgs = {}
+    global request_ids
+    request_ids = set()
+    for i in range(len(manifests)):
+        request_id = str(uuid.uuid4())
+        request_msgs[request_id] = gen_request_msg(
+            args=args,
+            digest=manifests[i],
+            reference=references[i],
+            request_id=request_id,
+        )
+        request_ids.add(request_id)
 
     umb.connect_and_subscribe(args.umb_listen_topic)
 
+    results = []
     try:
         retry_count = 3
         for i in range(retry_count + 1):
-            LOGGER.info(f"Sending signing request message...attempt #{i+1}")
-            umb.send(args.umb_publish_topic, json.dumps(request_msg))
+            LOGGER.info(
+                f"Sending {len(request_ids)} signing request messages...attempt #{i+1}"
+            )
+            for request_id in request_ids:
+                umb.send(args.umb_publish_topic, json.dumps(request_msgs[request_id]))
 
             wait_count = 0
-            LOGGER.debug(f"Checking for signing response result file {result_file}...")
-            while not os.path.exists(result_file):
-                time.sleep(WAIT_INTERVAL_SEC)
+            LOGGER.debug(
+                f"Checking for signing response result files with prefixes "
+                f"{request_ids}"
+            )
+            while len(request_ids) != 0:
                 wait_count += 1
                 if wait_count > TIMEOUT_COUNT:
                     LOGGER.warning("Timeout from waiting for signing response.")
                     break
+
+                time.sleep(WAIT_INTERVAL_SEC)
+
+                sig_received = set()
+                for request_id in request_ids:
+                    if os.path.exists(f"{request_id}-{result_file}"):
+                        with open(f"{request_id}-{result_file}", "r") as f:
+                            result_json = json.load(f)
+                            if result_json["signing_status"] == "success":
+                                results.append(result_json)
+                                sig_received.add(request_id)
+                            else:
+                                LOGGER.error(
+                                    f"Signing failure received for request {request_id}"
+                                )
```

Comment:
Is this really a signing failure or could it just be pending?

Reply:
So far the only two statuses for ISV signing are "success" and "failure". I can change the check to `!= "failure"` if we wanna be future-proof about adding more statuses.

Reply:
How about adding an `elif` condition to specifically check for the `failure` state? The `else` condition log message should then highlight the unknown state.

Reply:
Thanks, updated. Please check again.